test/smt/smt.test.ts

```diff
@@ -762,11 +766,12 @@ describe("Check SMT functionality via State", () => {
       await checkTestCase(testCaseEdge);
     });
 
-    it("Negative: add two leaves with maximum depth + 1", async () => {
-      await expect(state.migrateStateToSmt(genMaxBinaryNumber(31), 100, 0, 0)).not.to.be.reverted;
-      await expect(state.migrateStateToSmt(genMaxBinaryNumber(32), 100, 0, 0)).to.be.revertedWith(
-        "Max depth reached"
-      );
+    it.skip("Negative: add two leaves with maximum depth + 1", async () => {
+      await expect(state.migrateStateToSmt(genMaxBinaryNumber(31), 100, 0, 0))
+        .not.to.be.reverted;
+      await expect(
+        state.migrateStateToSmt(genMaxBinaryNumber(32), 100, 0, 0)
+      ).to.be.revertedWith("Max depth reached");
```

Comment:
The fact that you was forced to skip this test, looks like a result of wrong test organisation (brought by me earlier).
Better to reorg this tests in a way to test directly the SMT lib bypassing the State contract.